include/itkMorphologicalContourInterpolator.hxx

```diff
@@ -127,11 +162,255 @@ void MorphologicalContourInterpolator<TImage>::DetermineSliceOrientations()
   }
 }
 
+template<class TImage>
+typename TImage::Pointer MorphologicalContourInterpolator<TImage>::RegionedConnectedComponents
+(typename TImage::RegionType region, typename TImage::PixelType label, itk::IdentifierType &objectCount)
+{
+  typedef BinaryThresholdImageFilter<typename TImage, BoolImageType> BinarizerType;
+  BinarizerType::Pointer bin = BinarizerType::New();
+  bin->SetLowerThreshold(label);
+  bin->SetUpperThreshold(label);
+  bin->SetInput(this->GetInput());
```

Comment:
In GenerateData, graft the input to a new pointer, and operate on it as an input.  See

  http://review.source.kitware.com/#/c/19809/1/Modules/Filtering/ImageFeature/include/itkCannyEdgeDetectionImageFilter.hxx

for an example.

This prevents the pipeline update from propagating every time it is called.